### ISSUE TITLE
[client] — Build rate-limiting into the client

### ISSUE THREAD

**Issue by davesag** [sanity-io/sanity#1177]
I spent a fair bit of time today redoing some of my project code to allow for the rate limiting that got turned on last night.  I have no issue with rate limiting but it seems to me that it would be much more convenient, given it's such a cross-cutting concern, and can be non-trivial to implement, to just have the rate limiting built into the javascript client itself.  Then it's just not something that we devs need to worry our pretty little heads over.

**Comment by nicholasklem** [comment]
Agree. Escalated internally here.

**Comment by coreyward** [comment]
Any progress on this, @nicholasklem?

**Comment by nicholasklem** [comment]
No, unfortunately nothing yet. I've added your comment to our internal issue, maybe we can get some progress by making noise.

**Comment by simonhrogers** [comment]
+1

**Comment by good-idea** [comment]
Here to make some more noise!

**Comment by sc-searchengineunity** [comment]
currently working through this problem. +1 on request!

**Comment by simen** [comment]
Yes, this is something we need to do.

**Comment by KwehDev** [comment]
+1 For this, I'm surprised there didn't seem to be anything already built in.

**Comment by rexxars** [comment]
Moving this to the client repository ➡️ https://github.com/sanity-io/client
